numpy/lib/_shape_base_impl.py

```diff
@@ -1241,16 +1244,29 @@ def tile(A, reps):
 
     >>> b = np.array([[1, 2], [3, 4]])
     >>> np.tile(b, 2)
+    array([[1, 2, 1, 2],
+           [3, 4, 3, 4]])
+    >>> np.tile(b, 2, axis=-1)
```

Comment:
I would give your example with `axis=0` here instead, as that immediately shows the benefit.

And then perhaps show a new example with `np.tile(b, (2, 2))`

Reply:
This specific examples show that the default behavior is equivalent to `axis=-1`

Reply:
I think the examples should just show the "preferred" way to do things, not every possible way. And I think if we go for adding the `axis` argument, the preferred way would be to use an integer for repeat and the axis to indicate along which axis the repeat should be done.

I think it will be important that the examples show the benefits for suggesting this to the array API.